### PR TITLE
Load the credentials the same way as client go does

### PR DESCRIPTION
**Pull request by houshengbo**

### CODE REVIEW

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -105,11 +105,11 @@
 		},
 		{
 			"ImportPath": "github.com/apache/incubator-openwhisk-client-go/whisk",
-			"Rev": "b3d194d8b937a801ee11efcf7f2544d9cea1f58e"
+			"Rev": "8a672e5f39dd0bbdf6ca9460782557d7f449103c"
```
Comment (mrutkows) [inline Godeps/Godeps.json]: @houshengbo is there a way to automate this?  
AND is there a way for 'wskdeploy' to somehow say built on "Go Client version X"?

File: deployers/whiskclient.go [hunk 2]
```diff
@@ -28,113 +28,165 @@ import (
...
+    // First, we look up the above variables in the deployment file.
+    if utils.FileExists(deploymentPath) {
+        mm := parsers.NewYAMLParser()
+        deployment := mm.ParseDeployment(deploymentPath)
+        credential.Value = deployment.Application.Credential
+        credential.Source = DEPLOYMENTFILE
+        namespace.Value = deployment.Application.Namespace
+        namespace.Source = DEPLOYMENTFILE
+        baseUrl.Value = deployment.Application.BaseUrl
+        baseUrl.Source = DEPLOYMENTFILE
```
Comment (mrutkows) [inline deployers/whiskclient.go]: @houshengbo Where is the logic to pull 'credential' value from manifest.yaml if not found in deployment.yaml?

File: deployers/whiskclient.go [hunk 3]
```diff
@@ -28,113 +28,165 @@ import (
...
+    // Fourth, we look up the variables in whisk.properties on a local openwhisk deployment.
+    whiskproperty, _ := GetWskPropFromWhiskProperty(pi)
+    credential = GetPropertyValue(credential, whiskproperty.AuthKey, WHISKPROPERTY)
+    namespace = GetPropertyValue(namespace, whiskproperty.Namespace, WHISKPROPERTY)
+    baseUrl = GetPropertyValue(baseUrl, whiskproperty.WHISKAPIURL, WHISKPROPERTY)
+
+    // If we still can not find the variables we need, check if it is interactive mode. If so, we accept the input
+    // from the user. The namespace will be set to a default value, when the code reaches this line, because WSKPROPS
+    // has a default value for namespace.
+    if len(baseUrl.Value) == 0 && isInteractive == true {
```
Comment (mrutkows) [inline deployers/whiskclient.go]: @houshengbo Nice catch, forgot about interactive mode.

File: deployers/whiskclient.go (hunk 2 above)
Comment (houshengbo) [reply deployers/whiskclient.go]: so far it has not been read from manifest.yml.

File: Godeps/Godeps.json (hunk 1 above)
Comment (houshengbo) [reply Godeps/Godeps.json]: I have no idea about how to automate the version pinning. have to investigate more.

**Comment by houshengbo** [comment]
@mrutkows I have already added the manifest yml as the source of credentials.

**Comment by mrutkows** [comment]
Fixed with https://github.com/apache/incubator-openwhisk-wskdeploy/pull/392

**Pull request merged**